### DISCUSSION TITLE
managing cli commands like: make, terraform, terragrunt etc.

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#20016] [Suggest an Idea]
### Type of discussion.

I just want to chat

### Tell us more.

My automation requires various command line tools like mentioned in the subject. On top of that, I want to keep track of their versions. Is there a lock file I can use which is supported by Renovate?

**Comment by secustor** [comment]
If you want to manage tools used in a repository, the best supported way by Renovate currently is [`asdf`](https://github.com/asdf-vm/asdf). The docs page for the manager contains currently a list of supported tools: https://docs.renovatebot.com/modules/manager/asdf/

Alternatively you can use something similar to [tfenv](https://github.com/tfutils/tfenv) and take advantage of Regex managers

**Reply by lmilbaum** [reply]
Thanks @secustor. Interesting. I wasn't aware that those tools exist. Will look into it.
